seacrowd/sea_datasets/wikihow_gosc/wikihow_gosc.py

```diff
@@ -0,0 +1,138 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """
+@inproceedings{lyu-etal-2021-goal,
+    title = "Goal-Oriented Script Construction",
+    author = "Lyu, Qing  and
+      Zhang, Li  and
+      Callison-Burch, Chris",
+    editor = "Belz, Anya  and
+      Fan, Angela  and
+      Reiter, Ehud  and
+      Sripada, Yaji",
+    booktitle = "Proceedings of the 14th International Conference on Natural Language Generation",
+    month = aug,
+    year = "2021",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/2021.inlg-1.19",
+    doi = "10.18653/v1/2021.inlg-1.19",
+    pages = "184--200",
+}
+"""
+_LOCAL = False
+_LANGUAGES = {"ind": "id", "tha": "th", "vie": "vn"}
+_DATASETNAME = "wikihow_gosc"
+_DESCRIPTION = """
+This dataset consists of wikiHow goal-oriented scripts. For each goal or task, sections with steps to achieve this task are
+generated. Both the sections and steps within them are classified as either ordered or unordered.
+"""
+
+_HOMEPAGE = "https://github.com/veronica320/wikihow-GOSC/tree/main?tab=readme-ov-file"
+_LICENSE = Licenses.MIT.value
+_URL = "https://drive.google.com/uc?id=1AqAocrNFEPhBAfa5ATCj-3xMWbq659ME"
+
+_SUPPORTED_TASKS = [Tasks.INSTRUCTION_TUNING]
+_SOURCE_VERSION = "1.0.0"
+
+
+class WikiHowGOSCDataset(datasets.GeneratorBasedBuilder):
+    """Dataset of WikiHow tasks/goals with generated steps to perform them."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang}_source",
+            version=_SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema for {lang} language",
+            schema="source",
+            subset_id=f"{_DATASETNAME}_{lang}",
+        )
+        for lang in _LANGUAGES
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_ind_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        features = datasets.Features(
+            {
+                "title": datasets.Value("string"),
+                "category": datasets.Value("string"),
+                "sections": datasets.Sequence({"section": datasets.Value("string"), "steps": datasets.Sequence(datasets.Value("string")), "ordered": datasets.Value("int32")}),
+                "ordered": datasets.Value("int32"),
```

Comment:
hi @elyanah-aco, I checked this data and some of the examples doesn't have any `sections` key (probably a single-sectioned data).

This is an example of the key list of the data and (note: one-based index since I enumerate it from 1).
<img width="344" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/69744460/e1b7dd96-954b-403f-a3f6-4d51eed80251">
<img width="390" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/69744460/ea80afdc-b96f-404b-8ecd-b572cc2280ba">


can you make some readjustment on the `_generate_examples` to cater such cases? thx!

Reply:
@sabilmakbar I'm thinking like this:
```
"sections": ["section": "", "steps": ['" Mulailah membuat situs web.", "Gunakan Twitter."], "ordered": 1],
"ordered": 1
```
where `ordered` inside and outside `sections`  are the same. `ordered` inside sections indicates if steps inside that section are ordered, and `ordered` outside indicates if sections themselves are ordered. They should be the same if there's just one section

Reply:
I think for the outer `ordered`, it should be valued as 1, even if the inner section `ordered` value is 0 (the one coming from actual data) because it's a single-valued list anyway (indicates the list contents is ordered for `sections` data). The `ordered` in inner section (for `section` data, singleton list) will follow the initial `ordered` value from actual example. Wdyt?

If we're taking the example from prev ss (but the `ordered` value is 0 instead of 1), we can construct it like this:
```python
# the `ordered` value in sections list indicatest the actual `ordered` state on the `steps`
"sections": ["section": "", "steps": ['" Mulailah membuat situs web.", "Gunakan Twitter."], "ordered": 0],
"ordered": 1 # the value indicates the ordering in `sections` is ordered (which is 1 for singleton list)
```